src/Orleans/Runtime/GrainReference.cs

```diff
@@ -501,6 +503,23 @@ private static void CheckForGrainArguments(object[] arguments)
                     throw new ArgumentException(String.Format("Cannot pass a grain object {0} as an argument to a method. Pass this.AsReference<GrainInterface>() instead.", argument.GetType().FullName));
         }
 
+        /// <summary>
+        /// Sets target grain to the found instances of type GrainCancellationToken
+        /// </summary>
+        /// <param name="arguments"> Grain method arguments list</param>
+        /// <param name="target"> Target grain reference</param>
+        private static void SetGrainCancellationTokensTarget(object[] arguments, GrainReference target)
+        {
+            if (arguments == null) return;
+            foreach (var argument in arguments)
+            {
+                if (argument is GrainCancellationToken)
+                {
+                    ((GrainCancellationToken)argument).AddGrainReference(target);
+                }
```

Comment:
nit picking: to avoid the double cast (and reduce code), you could replace the if conditional with:
`(argument as GrainCancellationToken)?.AddGrainReference(target);`